### ISSUE TITLE
[BUG] Unable to refresh/enable JIT for apps on iOS/iPadOS 17

### ISSUE THREAD

**Issue by ShonP40** [SideStore/SideStore#374]
### Describe the bug

After updating to iOS & iPadOS 17, I've noticed that SideStore is no longer able to refresh apps (getting `The operation couldn’t be completed. Cannot start a misagent server on the device.` - `minimuxer.MinimuxerError error 18` & `The operation couldn’t be completed. Cannot start a AFC server on the device.` - `minimuxer.MinimuxerError error 14`)

### Instructions to reproduce

- Update to iOS or iPadOS 17 Developer Beta 1
- Open SideStore
- Try to refresh an app or enable JIT for an app

### What version of SideStore are you using?

0.4.0

**Comment by Spidy123222** [comment]
You need to do a new pairing file. Also jit won't work since we don't have developer disk image for iOS 17 and 16.6

**Comment by ShonP40** [comment]
oh right forgot I had to make a new one for every major update

Also, JIT was working fine for me on iPadOS 16.6 Beta 2

**Comment by ShonP40** [comment]
Well, I can't get it to ask me for a new pairing file.
Even tried installing an older version of SideStore

**Comment by ShonP40** [comment]
And now its crashing every few seconds of having the app open (not expired)

**Comment by jawshoeadan** [comment]
I've verified this issue. Debugging my current build of SideStore results in "dev-image-mounter (15): signal SIGABRT"

**Comment by Spidy123222** [comment]
Yeah we have verified the issue a while ago after beta release just been busy with other stuff. Will get fixed at some point but this is a dev beta so unsure if this will be consistent problem or not.

**Comment by WildRiley** [comment]
Can confirm.
Sidestore crashes less than 5 seconds after opening on iOS 17 developer beta.

**Comment by naturecodevoid** [comment]
iOS 17 has a new developer disk image system which has not yet been reverse engineered. JIT will not work, but the crashing should be temporarily fixed (no SideStore update needed, a permanent fix will be put in place in the future)

**Comment by ShonP40** [comment]
Xcode used to create disk images (DMGs) you could just copy into SideStore
But with Xcode 15 & iOS 17, it doesn’t seem to do that anymore

**Comment by naturecodevoid** [comment]
Xcode 17 has a new personalized disk image system. I believe pymobiledevice3 already has an implementation of it and the new remoted stuff, but it will most likely be a while before it's implemented in libimobiledevice and SideStore

**Comment by ShonP40** [comment]
Seems like they implemented it https://github.com/doronz88/pymobiledevice3/commit/0b19731c78f82528e78753a0a7fa765ec2adfaa6

But ye, will prob be a while until it gets added into `libmobiledevice` https://github.com/libimobiledevice/libimobiledevice/issues/1452

**Comment by heem898c** [comment]
Any news?

**Comment by naturecodevoid** [comment]
pymobiledevice developer has been making lots of progress reverse engineering the new protocol. We don't know when JIT will be working again though

**Comment by loyahdev** [comment]
> pymobiledevice developer has been making lots of progress reverse engineering the new protocol. We don't know when JIT will be working again though

Got any information on this I found someone who was able to mount the personalized image.

**Comment by naturecodevoid** [comment]
I believe pymobiledevice3 supports the new developer services. However, it's implemented in python which means that it most likely won't be usable in SideStore.

**Comment by loyahdev** [comment]
> I believe pymobiledevice3 supports the new developer services. However, it's implemented in python which means that it most likely won't be usable in SideStore.

Yeah I did see that when I was researching. Do you know of any implementations that allow jit to run on iOS 17? I've tried Xcode but since that debugs it doesn't give a benefit and apps run a lot slower showing more in emulation.

**Comment by Spidy123222** [comment]
> > I believe pymobiledevice3 supports the new developer services. However, it's implemented in python which means that it most likely won't be usable in SideStore.
> 
> Yeah I did see that when I was researching. Do you know of any implementations that allow jit to run on iOS 17? I've tried Xcode but since that debugs it doesn't give a benefit and apps run a lot slower showing more in emulation.

For Xcode you want to attach to process and then click detach to simulate what jit enablers do. I'm not sure if pymobiledevice has detach ability it should be like libimobiledevice which has it. it but should work as others have tried, when logging which is keeping it attached to computer it does makes things a bit slower. Depending on the app you enable that also has a factor on speed.

**Comment by realgamebreaker** [comment]
I have the same issue with my iPhone 6s running iOS 15.7.7

**Comment by ShonP40** [comment]
> I have the same issue with my iPhone 6s running iOS 15.7.7

That’s not the same issue, open a new [issue](https://github.com/SideStore/SideStore/issues)

**Comment by Spidy123222** [comment]
> I have the same issue with my iPhone 6s running iOS 15.7.7

If you have slow internet or having difficult time downloading from GitHub then it will likely fail. Others are able to use it on 15.7.7.

This is the dmg it downloads you can manually add it to sidestore. Install these then extract the zip file and change the name to 15.7.7.dmg and 15.7.7.dmg.signature into the sidestore directory under DMG folder (if it's not there then create the folder named that).

**Comment by ShonP40** [comment]
Seems like AltStore supports JITs on iOS 17 now
Any updates on when it’ll be available here?

https://fosstodon.org/@altstore/111048060197449142

**Comment by nythepegasus** [comment]
@ShonP40 Sadly their implementation is based off pymobiledevice3 which isn’t compiled, so we cannot use it on device. It also does not work for Windows or Linux since it’s based on pymobiledevice3. If you want JIT on device, stay on iOS 16 and lower for now until we figure something better out.

**Comment by ShonP40** [comment]
`pymobiledevice3` does support Windows & Linux though.
Also, iOS can run Python code (I already have a few apps that can do that).

**Comment by Spidy123222** [comment]
> `pymobiledevice3` does support Windows & Linux though.
> Also, iOS can run Python code (I already have a few apps that can do that).

Yes pymobiledevice3 supports windows and Linux. However the new functions for iOS/iPadOS 17 doesn't yet. To use pymobiledevice3 via sidestore would require modifying it and the effort for it would Be better suited to a native of a existing library format we already have and are doing.

**Comment by ShonP40** [comment]
ah, alright

**Comment by SARankDirector-Minecraft** [comment]
Could it be possible to implement a JITstreamer-like solution using a macOS vm or Docker-OSX? That might be a good stopgap until libimobiledevice updates.

**Comment by naturecodevoid** [comment]
We discussed this, if it's impossible to do it on device we will most likely make JITStreamer 2.0. However, jkcoxson said that it should be possible on-device.

**Comment by cottontails2** [comment]
Any news?

**Comment by sirpeep** [comment]
Having same issue, what does altstore use for jit?

**Comment by chuck-sudo** [comment]
Anyone have a solution for this that does not require Xcode? I don’t have a Mac.

**Comment by abhishekgi** [comment]
Is is possible for windows users to enable JIT by installing macos on windows with virtual box ?

**Comment by ThatRedstoneGuru** [comment]
I was going to try this, but my computer hates virtual machines. My PC corrupts it every time it’s in the setup.

**Comment by abhishekgi** [comment]
> I was going to try this, but my computer hates virtual machines. My PC corrupts it every time it’s in the setup.

I have successfully installed macos big sur on windows with help of VMware workstation pro 
I am also able to connect my iPad to virtual machine and iPad it also detected by altserver , it also syncs over wi fi I have installed altserve with macos and I have deleted iTunes and iCloud and have cleared all the unnecessary files of apple 🍎 I have followed the instructions to enable jit for iOS 17 but was stuck in 4 th command after a lot of process, there is a waring that it is not supported by macos 11 so now I am downloading macos Sonoma and I am pretty sure that I will make it

**Comment by Marat6** [comment]
Проблема с Jit не решается в IOS 17.2 beta 2?

**Comment by ryuya0124** [comment]
> Windows ユーザーが仮想ボックス上の Windows に MacOS をインストールすることで、iOS 17 で JIT を有効にすることは可能ですか?

I have started MacOS Sonoma on VMWare and the JIT activation was successful.

**Comment by abhishekgi** [comment]
> > Windows ユーザーが仮想ボックス上の Windows に MacOS をインストールすることで、iOS 17 で JIT を有効にすることは可能ですか?
> 
> I have started MacOS Sonoma on VMWare and the JIT activation was successful.

On iOS 17 ?

**Comment by HamzaTGP** [comment]
> > Windows ユーザーが仮想ボックス上の Windows に MacOS をインストールすることで、iOS 17 で JIT を有効にすることは可能ですか?
> 
> 
> 
> I have started MacOS Sonoma on VMWare and the JIT activation was successful.

I would like to report also success on a QEMU/KVM MacOS Sonoma virtual machine. I was able to activate JIT using AltStore. Still not a convenient solution tho

**Comment by MarkGG8181** [comment]
> Xcode 15 has a new personalized disk image system. I believe pymobiledevice3 already has an implementation of it and the new remoted stuff (I'm not sure if they have it fully working, but they have made the most progress), but it will most likely be a while before it's implemented in libimobiledevice and SideStore
> 
> In earlier versions of Xcode, the process of getting developer disk images were similar to dumping keys from a Nintendo switch. The images were uploaded to GitHub by various people, where they could be used by SideStore

any updates?

**Comment by abhishekgi** [comment]
Is jit streamer 2 possible?

**Comment by justandree** [comment]
Any news or updates?

**Comment by nythepegasus** [comment]
@justandree Best we've got for Windows, Linux, and macOS is pymobiledevice3 but it's still wired (macOS with Xcode is the only current way to wirelessly enable JIT). Here is my Windows guide, but macOS would be similar:
https://gist.github.com/nythepegasus/78c4f1474367a0fd4eef4c55a06cf919

Unsure on how to set the Linux patches up, might require compiling the kernel.

**Comment by justandree** [comment]
> @justandree Best we've got for Windows, Linux, and macOS is pymobiledevice3 but it's still wired (macOS with Xcode is the only current way to wirelessly enable JIT). Here is my Windows guide, but macOS would be similar: https://gist.github.com/nythepegasus/78c4f1474367a0fd4eef4c55a06cf919
> 
> Unsure on how to set the Linux patches up, might require compiling the kernel.

Found this giude for JIT on windows, for me it's simpler than yours. https://github.com/fritzlb/iOS17-JIT-WIN 
Still a PC is required for this solution, I hope someday there will be a new solution I can use anywhere. (Maybe with the old methos of keeping the app pressed on SideStoreand enabling it from there.)

**Comment by nythepegasus** [comment]
The reason I relabeled this as a feature request is because to do all of that on device would require a ton of research into how QUIC works as well as the new personalized disk images. It's not as easy as it was on iOS 16 and lower to enable JIT where you could just mount any valid developer disk image over the old API, no personalization. iOS 17 got rid of the old way of doing it, so a computer will be required for the foreseeable future (again, I could be wrong, but on device enabling on iOS 17+ will be a ways away in SideStore unless things change).

**Comment by jkcoxson** [comment]
Will be fixed in #953
